Sources/System/FilePath/FilePathString.swift

```diff
@@ -96,6 +148,34 @@ extension FilePath.Root {
     self.init(_platformString: platformString)
   }
 
+  public init?(platformString: [CInterop.PlatformChar]) {
+    if let _ = platformString.firstIndex(of: 0),
+       let component = platformString.withUnsafeBufferPointer({
+         FilePath.Root(_platformString: $0.baseAddress!)
+       }) {
+      self = component
+      return
+    }
+    return nil
+  }
+
+  public init?(platformString: inout CInterop.PlatformChar) {
+    guard platformString == 0 else { return nil }
+    let root = withUnsafePointer(to: &platformString) {
+      FilePath.Root(_platformString: $0)
+    }
+    guard let root = root else { return nil }
+    self = root
+  }
+
+  public init?(platformString: String) {
```

Comment:
What calls this?

Reply:
That's another valid pointer conversion; ideally we'd mark this as deprecated. We can also not bother with it.

Reply:
The ideal outcome is to restrict pointer conversions to C interop only, and add an attribute to opt in to pointer conversions for shims that call out to C functions. Obviously that's not happening this year.